### PR TITLE
Add support for textDecorationLine property

### PR DESCRIPTION
**Pull request by davidseik**
Hi, this adds support for the textDecorationLine property for text.
Tested on ***Sketch 44.1***
Example of usage:
```js
<Text  style={{textDecorationLine: 'underline'}}>
  Hello World
</Text>
```

which outputs this:
screen shot 2017-07-07 at 11 13 54 am

The API supports the following attributes: `none`, `underline`, `line-through`, `double`.

I have some questions though:
* Should I create some tests around this?
* How about the documentation, does this grant an update?
* There was an autoformat on commit, I didn't make some of those changes. is that okay?

### CODE REVIEW

**Comment by jongold** [comment]
Wooo, great work @davidseik :)

- we haven't found a good solution for tests right now, unfortunately. we're hoping that relying on small functions & flow keeps this mostly error free, but we just test in `examples/` 😱 
- Yes! we can add the textDecoration* props to the Styling documentation :)
- That's fine! Prettier is set up to automatically format files; presumably that file just hadn't been touched since we added it :)

Will dig in this afternon

File: src/buildTree.js [hunk 1]
```diff
@@ -12,13 +12,14 @@ const INHERITABLE_STYLES = [
   'fontSize',
   'fontStyle',
   'fontWeight',
+  'textAlign',
+  'textDecoration',
```
Comment (jongold) [inline src/buildTree.js]: we mirror the React Native APIs as much as possible — can you change this to `textDecorationLine` to match?

Comment (davidseik) [inline src/buildTree.js]: Hi Jon, I actually wanted to use the exact same API of textDecorationLine but I found this:
https://github.com/airbnb/react-sketchapp/blob/master/src/stylesheet/expandStyle.js#L61

When the style gets passed down as a textStyle this part expands it to be renamed `textDecoration`; I don't know why this is needed. I can go ahead and remove the shortHand here and add the `textDecorationLine` over here so it's propagated like it. Thoughts?

Comment (jongold) [inline src/buildTree.js]: 🙃 I think maybe it was actually correct the first time - looks like I have your original commit and it's indeed working properly by expanding the textDecorationLine into textDecoration. Not sure what went wrong the first time I reviewed it, sorry!

Documentation update looks good - we want the public interface to be `textDecorationLine`.

Comment (davidseik) [inline src/buildTree.js]: Got it, then let me change it back.

Comment (davidseik) [inline src/buildTree.js]: Funny thing is that because of the expansion they both work `textDecoration` & `textDecorationLine`; Thoughts?

**Comment by davidseik** [comment]
@lelandrichardson @BenjaminCorey can you guys check if this can be merged?

**Pull request merged**